aptos-move/framework/aptos-framework/sources/GovernanceProposal.move

```diff
@@ -3,16 +3,44 @@
 /// This is separate from the AptosGovernance module to avoid circular dependency between AptosGovernance and Stake.
 module AptosFramework::GovernanceProposal {
     friend AptosFramework::AptosGovernance;
+    use Std::ASCII::{String, string};
 
-    struct GovernanceProposal has store, drop {}
+    struct GovernanceProposal has store, drop {
+        /// The location (e.g. url) where the proposal resolution script's code can be accessed.
+        /// Maximum length allowed is 256 chars.
+        code_location: String,
```

Comment:
will these fields be used on-chain?

Reply:
No I don't think so. From your experience with the Token design, does it make sense to have these on or off chain?

Reply:
most of the fields stored on-chain can be used on-chain. We can store on-chain if this is easier for now.  If you think these field wouldn't be use on-chain, try to emit in events and use indexer eventually

Reply:
Makes sense. I think for now, it's hard to support indexer with the UIs we have.